src/main/java/core/basesyntax/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Car;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+
+    @Override
+    public Car create(Car car) {
+        Storage.addCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        return getAll().stream()
+                .filter(car -> Objects.equals(car.getId(), id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return Storage.cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        return Storage.cars.set(Storage.cars.indexOf(car), car);
```

Comment:
if car is not present in the list index will be -1 and you will get here IndexOutOfBound which is not an expected behavior for the update method.

Reply:
Fixed